Holovibes/sources/core/pipe.cc

```diff
@@ -517,29 +457,34 @@ void Pipe::insert_filter2d_view()
     }
 }
 
+template <typename SettingType>
+cudaMemcpyKind Pipe::get_memcpy_kind()
```

Comment:
I'm not sure that this function makes the code clearer to read. Even if it remove some boilerplate. Maybe give a name clearer or do another function for the else cases when it is used.

Reply:
I am saying this for examples like :
if (setting<settings::InputQueueLocation>() == Device::GPU)
                memcpy_kind = get_memcpy_kind<settings::RawViewQueueLocation>();
            else
                memcpy_kind = setting<settings::RawViewQueueLocation>() == Device::GPU ? cudaMemcpyHostToDevice
                
 Where you use the function for the first line but let the 2nd like this. I think get_memcpy_kind name do not give enough info since the line in the else is also getting the kind of memcpy